Samples/DynamicDependencies/DirectX/direct3d9ex/BackgroundThread.cpp

```diff
@@ -0,0 +1,380 @@
+﻿//--------------------------------------------------------------------------------------
+// File: BackgroundThread.cpp
+//
+// Sample showing how to use D3D9Ex advanced features.
+//
+// Copyright(c) Microsoft Corporation.
+// Licensed under the MIT License.
+//--------------------------------------------------------------------------------------
+#include <windows.h>
+#include <d3d9.h>
+#include <d3dx9.h>
+
+#define SAFE_RELEASE(p) { if(p) { (p)->Release(); (p) = NULL; } }
+
+//-----------------------------------------------------------------------------
+// structures
+//-----------------------------------------------------------------------------
+struct BACKGROUNDVERTEX
+{
+    FLOAT x, y, z;	  // The untransformed position for the vertex
+    DWORD dwColor;	  // Color
+};
+#define D3DFVF_BACKGROUNDVERTEX (D3DFVF_XYZ|D3DFVF_DIFFUSE)
+
+//--------------------------------------------------------------------------------------
+// Global variables
+//--------------------------------------------------------------------------------------
+bool g_bEndThread = false;
+IDirect3DTexture9*	g_pRenderTarget = NULL;
```

Comment:
raw COM pointers are dangerous, use wil::com_ptr, winrt::com_ptr, or anything but a raw type.

Reply:
see comment above about taking an older existing sample from dx9 and showing how to use dynamic dependencies to hook up to the MSIX DirectX framework package.